samples/snippets/src/main/java/com/example/spanner/HubbleTransactionsCodeLab.java

```diff
@@ -290,6 +242,175 @@ public void updatesAndReads(
         NUM_PROCESSORS / 2);
   }
 
+  public void doWorkSingleTransactionParallelLocking(DatabaseClient dbClient) {
+    // getting the available processor
+    ExecutorService executorService = Executors.newFixedThreadPool(NUM_PROCESSORS);
+
+    List<Future<Integer>> futures = new ArrayList<>();
+    // List to store the count of total number of workItem(row) which has not yet completed means
+    // is_done is false.
+    List<String> workItemCount = new ArrayList<>();
+    // Below logic is reading the generated_value for the rows which are having is_done=false. We
+    // have added generated_value column value in incremented order starting from 1 in the table.
+    // This will help us to assign executing the random workItem(rows) work without overlapping. As
+    // each thread will pick up different generated_value to query.
+    try (ResultSet resultSet =
+        dbClient
+            .singleUse() // Execute a single read or query against Cloud Spanner.
+            // Below query can be parameterized to but kept here for simplicity
+            .executeQuery(
+                Statement.of("SELECT generated_value FROM WorkList WHERE is_done is false"))) {
+      while (resultSet.next()) {
+        workItemCount.add(resultSet.getString(0));
```

Comment:
This, in general, is not a pattern we should use. What if the workItem set is being updated? What if the workItem set doesn't fit into local memory? While this works in our toy example, it is very unlikely to work in a real world use case.